.github/workflows/version-code-and-release.yml

```diff
@@ -0,0 +1,59 @@
+name: Version Code Check and Draft Release
+
+on:
+  pull_request:
+    paths:
+      - 'lib/kql/**'
+      - 'lib/kibana/**'
+      - 'hunting/**'
+      - 'etc/**'
+      - 'pyproject.toml'
+    types: [opened, reopened, synchronize]
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  version_check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Check for changes in kql, kibana, hunting, and etc
+        id: check_changes
+        run: |
+          CHANGED_FILES=$(git diff --name-only ${{ github.event.before }} ${{ github.sha }} | grep -E 'lib/kql|lib/kibana|detection_rules|tests|hunting|etc/' || echo "no-changes")
+          if [ "$CHANGED_FILES" != "no-changes" ]; then
+            echo "CHANGES_FOUND=true" >> $GITHUB_ENV
+          else
+            echo "CHANGES_FOUND=false" >> $GITHUB_ENV
+          fi
+
+      - name: Fail if no version bump in pyproject.toml
+        if: env.CHANGES_FOUND == 'true'
+        run: |
+          if ! git diff --name-only ${{ github.event.before }} ${{ github.sha }} | grep 'pyproject.toml'; then
+            echo "Code changes detected in but core pyproject.toml was not updated."
```

Comment:
We should remember that the requirement to update the version in every pull request that touches code will introduce race conditions/conflicts between concurrent pull requests.

Reply:
Good point. We dont use merge queues so I dont think it'll be an issue.